### ISSUE TITLE
Cannot read property 'apply' of undefined

### ISSUE THREAD

**Issue by alfredwesterveld** [codecept-js/mock-request#4]
Update:

```
I figured out why this was causing issues. You were using puppeteer v2.2.1 and then base script also not fails for me.
```

Hello thank you for this project. When I run with following settings:

```
const { setHeadlessWhen } = require('@codeceptjs/configure');

// turn on headless mode when running with HEADLESS=true environment variable
// HEADLESS=true npx codecept run
setHeadlessWhen(process.env.HEADLESS);

exports.config = {
  tests: './test/*_test.js',
  output: './test/output',
  helpers: {
    MockRequestHelper: {
      require: '@codeceptjs/mock-request',
      mode: 'replay',
    },
    Puppeteer: {
      url: 'https://www.google.com',
      show: true,
      windowSize: '',
    }
  },
  include: {
    I: './steps_file.js'
  },
  bootstrap: null,
  mocha: {},
  name: 'play-ception',
  plugins: {
    retryFailedStep: {
      enabled: true
    },
    screenshotOnFail: {
      enabled: true
    }
  }
}
```

the following simple test:

```
Feature('google');

Scenario('test something', async (I) => {
    I.amOnPage(
        'https://www.google.com'
    );
    I.startMocking();
});
```

I get the following error:

```
  TypeError: Cannot read property 'apply' of undefined
      at PuppeteerCodeceptJSAdapter._callListenersWith (node_modules/@pollyjs/adapter-puppeteer/dist/cjs/pollyjs-adapter-puppeteer.js:2418:28)
      at PuppeteerCodeceptJSAdapter.attachToPageEvents (node_modules/@pollyjs/adapter-puppeteer/dist/cjs/pollyjs-adapter-puppeteer.js:2328:10)
      at PuppeteerCodeceptJSAdapter.onConnect (node_modules/@pollyjs/adapter-puppeteer/dist/cjs/pollyjs-adapter-puppeteer.js:2233:10)
      at PuppeteerCodeceptJSAdapter.connect (node_modules/@pollyjs/adapter/dist/cjs/pollyjs-adapter.js:1130:12)
      at Polly.connectTo (node_modules/@pollyjs/core/dist/cjs/pollyjs-core.js:6430:13)
      at config.adapters.forEach.adapter (node_modules/@pollyjs/core/dist/cjs/pollyjs-core.js:6315:50)
      at Array.forEach (<anonymous>)
      at Polly.configure (node_modules/@pollyjs/core/dist/cjs/pollyjs-core.js:6315:26)
      at new Polly (node_modules/@pollyjs/core/dist/cjs/pollyjs-core.js:6216:10)
      at PuppeteerConnector.connect (node_modules/@codeceptjs/mock-request/connector/Puppeteer.js:43:18)
```

following versions:
```
CodeceptJS v2.6.7
puppeteer 4.0.1 (also 5)
```

**Comment by sunilbatchu** [comment]
I got the same error. 
Thanks a lot for the update. the above error is fixed after changing puppeteer version as you mentioned.

However it doesn't record xhr request anywhere. 
Could you please help me the configuration for record/replay for xhr requests

**Comment by alfredwesterveld** [comment]
> I got the same error.
> Thanks a lot for the update. the above error is fixed after changing puppeteer version as you mentioned.
> 
> However it doesn't record xhr request anywhere.
> Could you please help me the configuration for record/replay for xhr requests

For now I got it working by installing `puppeteer@2.2.1`

**Comment by serzhshakur** [comment]
This is fixed in latest pollyjs version. Check [this thread](https://github.com/codeceptjs/CodeceptJS/issues/2700) for more details

**Comment by davidpurkiss** [comment]
Is this really fixed though? Using `https://github.com/rogeriochaves/npm-force-resolutions` or yarn the equivalent doesn't seem like a good fix and this package's dependencies should be updated. E.g. `npm-force-resolutions` doesn't work in our build environment due to auth issues with NPM.

**Comment by davidpurkiss** [comment]
Also, there is an [explicit warning](https://github.com/rogeriochaves/npm-force-resolutions#warning-before-you-start) on the `npm-force-resolutions` readme that states,

>The use case for this is when there is a security vulnerability and you MUST update a nested dependency otherwise your project would be vulnerable. But this should only be used as a last resource, you should first update your top-level dependencies and file an issue for them to update the vulnerable sub-dependencies (npm ls <vulnerable dependency> can help you with that).

**Comment by connected** [comment]
MockRequest is a must have feature and its a shame that it doesn't work out of the box. Please update the deps.

**Comment by rhys-thomas-QA** [comment]
Still having the issue above in October 2021, would be a real nice feature to have

**Comment by DavertMik** [comment]
Fixed in 0.3.1
